### ISSUE TITLE
[Bug] Database query not running on page load

### ISSUE THREAD

**Issue by philipnye** [appsmithorg/appsmith#7982]
## Description

MSSQL database connection query does not run on page load despite 'Run query on page load' option being set.

Query only runs when some element of the app is interacted with (e.g. moving between pages of the app; updating an input that has onOptionChange connected to the query).

https://www.loom.com/share/878644907b384df88c0f3a98f6cc2134

### Important Details

- Version: Cloud
- OS: Windows 10
- Browser: Chrome
- Environment: Production

**Comment by Nikhil-Nandagopal** [comment]
@philipnye Apologies for the inconvenience. Thank you for reporting this issue!
@nidhi-nair can you please have a look?

**Comment by trishaanand** [comment]
@philipnye : Quick question. When you say that when you move between pages of the application, and you move to this particular page, does the query get executed? I ask this because you said `Query only runs when some element of the app is interacted with (e.g. moving between pages of the app; `. That in essence is the query executing on page load. Is the experience that sometimes when you open the said page the query doesnt run?

**Comment by philipnye** [comment]
Hi @trishaanand, thanks. The table that is connected to the query is on the home page of my app - when I open the app the query is being executed 0% of the time. But when I move to a different page of the app then move back to the home page the query seems to execute 100% of the time. So there seems to be something wrong specifically with the initial page load.

**Comment by nidhi-nair** [comment]
Hi @philipnye ! Would you mind sharing the configuration for this query? Or whether it expects any variables? You can share this info with me on Discord @ `hawk3y3#2925`

**Comment by nidhi-nair** [comment]
@trishaanand this is another instance of #7065. This series of dependencies has a multiselect in between that is breaking the chain.

@philipnye In the mean time, you could work around this bug that Trisha will be fixing by triggering the query using a button. Does that work?

**Comment by philipnye** [comment]
@nidhi-nair Yes, I can use that workaround in the meantime. Thanks very much for your help identifying the issue.

**Comment by prapullac** [comment]
MSSQL database connection credentials required
